test/integration/volume/attach_detach_test.go

```diff
@@ -321,6 +330,19 @@ func waitForPodsInDSWP(t *testing.T, dswp volumecache.DesiredStateOfWorld) {
 	}
 }
 
+// wait for pods to be observed in desired state of world
+func waitForPodFuncInDSWP(t *testing.T, dswp volumecache.DesiredStateOfWorld, checkTimeout time.Duration, failMessage string, tpodFunc podCountFunc) {
```

Comment:
instead of passing in podCountFunc, pass in the number of pods you want to wait for.